### PR TITLE
[WIP] Move Knative Prow config files from GoogleCloudPlatform

### LINKED ISSUE #2982: Move Knative Prow config files from GoogleCloudPlatform/oss-test-infra repo to the knative/test-infra repo
As the outcome of [Proposal: Reducing Productivity Oncall Duties](https://docs.google.com/document/d/1tktwX9qHR9DKiA6yZCEMRYouNtScyRVW4ztVea-MJ1M/edit?resourcekey=0-G8IkKBaNP2s7xc2j92nvAQ#), Knative Prow config files were moved from GoogleCloudPlatform/oss-test-infra repo to the knative/test-infra repo, which are managed and maintained by the Prow team at Google.

Now that Knative has been donated to CNCF, we'd want to move the Knative Prow to be community-managed. This issue is for tracking the effort on moving the Prow config files back to the knative/test-infra repo and configuring autobump.

The steps will be:
1. Move everything from https://github.com/GoogleCloudPlatform/oss-test-infra/tree/master/prow/knative, which includes Prow deployment config files and some other script files.
2. Move the autobump Prow job from https://github.com/GoogleCloudPlatform/oss-test-infra/blob/master/prow/prowjobs/GoogleCloudPlatform/oss-test-infra/gcp-oss-test-infra-config.yaml#L167-L239, which is for automatically creating PRs to bump the image versions for Prow.
> Note there is also an autobump Prow job in the knative/test-infra repo (https://github.com/knative/test-infra/blob/main/config/prod/prow/knative-autobump-config.yaml), so we may consider combining them.
3. Move the postsubmit Prow job from https://github.com/GoogleCloudPlatform/oss-test-infra/blob/master/prow/prowjobs/GoogleCloudPlatform/oss-test-infra/gcp-oss-test-infra-config.yaml#L26-L58 to auto-deploy Prow.

More details can be found in the [Knative test infrastructure 1.0](https://docs.google.com/document/d/1U6wybYjCslEhsKOiennrwhoKZmSVnd1NGbFI8gnnnOg/edit#) doc

### PR DESCRIPTION
**Pull request by kvmware**
Fixes #2982 

Beginning effort to move prow under community management.

### CODE REVIEW

File: config/prow/Makefile.gcloud.mk [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Copyright 2019 Google LLC
```
Comment (kvmware) [inline config/prow/Makefile.gcloud.mk:1]: With the exception of `save-kubeconfig` this seems to be doing what https://github.com/knative/test-infra/blob/main/config/Makefile does...

File: config/prow/build-cluster/README.md [hunk 2]
```diff
@@ -1,7 +1,11 @@
+## Knative prow
+
+This directory contains prow configs hosted for Knative. This prow is bumped by
+[`bump.sh`](../bump.sh) script
```
Comment (kvmware) [inline config/prow/build-cluster/README.md:4]: is it though?

File: config/prow/build-cluster/README.md [hunk 3]
```diff
@@ -29,3 +30,40 @@ Copy secrets from old cluster to build cluster:
...
+  - This is a very basic cluster with important secrets
+
+### Manually Deploy
+
+Manual deployments are defined as in [Makefile](./Makefile), specifically:
+
+- `make -C prow/knative deploy`: deploys all yamls under [cluster](./cluster)
+- `make -C prow/knative deploy-build`: deploys all yamls under [cluster/build](./cluster/build)
+
+## Prow Secrets
```
Comment (kvmware) [inline config/prow/build-cluster/README.md:52]: There is a section above secrets. Should that be merged here?

File: config/prow/build-cluster/create-build-cluster-helper.sh [hunk 4]
```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
```
Comment (kvmware) [inline config/prow/build-cluster/create-build-cluster-helper.sh:1]: This should probably just be merged with the existing `create-build-cluster.sh` file.

File: config/prow/build-cluster/create-build-cluster.sh [hunk 5]
```diff
@@ -33,4 +33,5 @@ export GCSBUCKET="${GCSBUCKET:-knative-prow}"
 export NODECOUNT=4
 export OUT_FILE="build-cluster-kubeconfig.yaml"
 
-bash <(curl -sSfL https://raw.githubusercontent.com/GoogleCloudPlatform/oss-test-infra/master/prow/create-build-cluster.sh) "$@"
+#bash <(curl -sSfL https://raw.githubusercontent.com/GoogleCloudPlatform/oss-test-infra/master/prow/create-build-cluster.sh) "$@"
+source create-build-cluster-helper.sh
```
Comment (kvmware) [inline config/prow/build-cluster/create-build-cluster.sh:37]: Should just be inlined.

File: config/prow/jobs/custom/prow-autobump.yaml [hunk 6]
```diff
@@ -0,0 +1,74 @@
+periodics:
+- cron: "15 15-23 * * 1-5"  # Run at 7:15-15:15 PST (15:15 UTC) Mon-Fri
+  name: ci-oss-test-infra-autobump-knative-prow
+  cluster: test-infra-trusted
+  decorate: true
+  extra_refs:
+  - org: GoogleCloudPlatform
```
Comment (kvmware) [inline config/prow/jobs/custom/prow-autobump.yaml:7]: should be `knative` ?

File: config/prow/jobs/custom/prow-autobump.yaml [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+    repo: oss-test-infra
```
Comment (kvmware) [inline config/prow/jobs/custom/prow-autobump.yaml:8]: s/oss-//

File: config/prow/jobs/custom/prow-autobump.yaml [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+    base_ref: master
+  annotations:
+    testgrid-dashboards: googleoss-test-infra
```
Comment (kvmware) [inline config/prow/jobs/custom/prow-autobump.yaml:11]: Probably should go under a `prow` dashboard or `autobump` or something.. there is a `utilities` dashboard, but don't know if that is the right place.

File: config/prow/jobs/custom/prow-autobump.yaml [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+        readOnly: true
+      - name: ssh
+        mountPath: /root/.ssh
+    volumes:
+    - name: github
+      secret:
+        secretName: oauth-token-2
+    - name: ssh
+      secret:
+        secretName: google-oss-robot-ssh-keys
```
Comment (kvmware) [inline config/prow/jobs/custom/prow-autobump.yaml:34]: this probably needs to be changed?

File: config/prow/jobs/custom/prow-deploy.yaml [hunk 10]
```diff
@@ -0,0 +1,35 @@
+postsubmits:
+  knative/test-infra:
+  - name: post-oss-test-infra-deploy-knative-prow
+    cluster: test-infra-trusted
+    run_if_changed: '^prow/knative/cluster/|^prow/oss/serviceaccounts/'
+    decorate: true
+    branches:
+    - master
+    annotations:
+      testgrid-dashboards: googleoss-test-infra
```
Comment (kvmware) [inline config/prow/jobs/custom/prow-deploy.yaml:10]: again this should probably go under `prow` or something

File: config/prow/pj-on-kind.sh.from.oss.repo [hunk 11]
```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
```
Comment (kvmware) [inline config/prow/pj-on-kind.sh.from.oss.repo:1]: This an the original (?) https://github.com/knative/test-infra/blob/main/config/prow/pj-on-kind.sh should be merged.

File: config/prow/Makefile.gcloud.mk (hunk 1 above)
Comment (chizhg) [inline config/prow/Makefile.gcloud.mk:1]: Please change the Copyright header to Knative, and elsewhere.

Comment (chizhg) [reply config/prow/Makefile.gcloud.mk:1]: Yeah we can probably just get rid of this file. I don't have a strong opinion about the `save-kubeconfig` target.

File: config/prow/README.md [hunk 12]
```diff
@@ -3,10 +3,8 @@
 This directory contains the config for our
 [Prow](https://github.com/kubernetes/test-infra/tree/master/prow) instance.
 
-- `Makefile` Commands to interact with the Prow instance regarding configs and
-  updates. Run `make help` for assistance.
-- `cluster/*.yaml` Deployments of the Prow cluster.
-- `core/*.yaml` Generated core configuration for Prow.
+- `../Makefile` Commands to interact with the Prow instance regarding configs and updates. Run `make help` for assistance.
```
Comment (chizhg) [inline config/prow/README.md:6]: It should be cleaner if we just move the `Makefile` to the `prow` folder. Sorry I missed that in my previous change.

File: config/prow/README.md [hunk 13]
```diff
@@ -3,10 +3,8 @@
...
+- `build-cluster/cluster/*.yaml` Deployments of the Prow cluster.
```
Comment (chizhg) [inline config/prow/README.md:7]: The layout in https://github.com/GoogleCloudPlatform/oss-test-infra/tree/master/prow/knative/cluster actually makes more sense so I would recommend directly porting them over here, and merge the existing config files in knative/test-infra into that layout. 

Let me know if you have questions regarding this layout, and I can walk you through it via video chat.

File: config/prow/build-cluster/README.md (hunk 2 above)
Comment (chizhg) [reply config/prow/build-cluster/README.md:4]: That script has been been deprecated for a long while. Currently they are using https://github.com/GoogleCloudPlatform/oss-test-infra/blob/master/prow/knative/knative-autobump-config.yaml which is backed by https://github.com/kubernetes/test-infra/tree/master/prow/cmd/generic-autobumper

File: config/prow/build-cluster/README.md (hunk 3 above)
Comment (chizhg) [reply config/prow/build-cluster/README.md:55]: Yes, let's also port them over. This aligns well with the longer term plan to also manage all the secrets in declarative yaml files. 
cc @upodroid

**Comment by kvmware** [comment]
Closing this in favor of https://github.com/knative/test-infra/pull/3069 which will have cleaner/easier to follow commits.

**Pull request closed without merging**